pinot-common/src/test/java/org/apache/pinot/common/audit/AuditUrlPathFilterTest.java

```diff
@@ -0,0 +1,167 @@
+
+  @Test
+  public void testMultiplePatternsCommaSeparated() {
+    String patterns = "health,api/users,admin";
+
+    assertThat(_filter.isExcluded("health", patterns)).isTrue();
+    assertThat(_filter.isExcluded("api/users", patterns)).isTrue();
+    assertThat(_filter.isExcluded("admin", patterns)).isTrue();
+    assertThat(_filter.isExcluded("metrics", patterns)).isFalse();
+  }
+
+  @Test
+  public void testMultiplePatternsWithTrimmingAndEmptyElements() {
+    String patterns = " health , , api/users , , ";
+
+    assertThat(_filter.isExcluded("health", patterns)).isTrue();
+    assertThat(_filter.isExcluded("api/users", patterns)).isTrue();
+    assertThat(_filter.isExcluded("metrics", patterns)).isFalse();
+  }
+
+  @Test
+  public void testAnyPatternMatchesReturnsTrue() {
+    String patterns = "nonexistent1,health,nonexistent2";
+
+    assertThat(_filter.isExcluded("health", patterns)).isTrue();
+    assertThat(_filter.isExcluded("nonexistent1", patterns)).isTrue();
+    assertThat(_filter.isExcluded("other", patterns)).isFalse();
+  }
+
+  // ===== Prefix Handling Tests =====
+
+  @Test
+  public void testAutomaticGlobPrefixAddition() {
+    assertThat(_filter.isExcluded("health", "health")).isTrue();
+    assertThat(_filter.isExcluded("api/users", "api/*")).isTrue();
+  }
+
+  @Test
+  public void testExplicitGlobPrefix() {
+    assertThat(_filter.isExcluded("health", "glob:health")).isTrue();
+    assertThat(_filter.isExcluded("api/users", "glob:api/*")).isTrue();
+  }
+
+  @Test
+  public void testExplicitRegexPrefix() {
+    String pattern = "regex:api/v[0-9]+/.*";
+    assertThat(_filter.isExcluded("api/v1/users", pattern)).isTrue();
+    assertThat(_filter.isExcluded("api/v123/anything", pattern)).isTrue();
+    assertThat(_filter.isExcluded("api/va/users", pattern)).isFalse();
+  }
+
+  @Test
+  public void testMixedPrefixes() {
+    String patterns = "glob:health,regex:api/v[0-9]+/.*,admin";
+
+    assertThat(_filter.isExcluded("health", patterns)).isTrue();
+    assertThat(_filter.isExcluded("api/v1/users", patterns)).isTrue();
+    assertThat(_filter.isExcluded("admin", patterns)).isTrue();
+  }
+
+  // ===== Error Handling Tests =====
+
+  @Test
+  public void testInvalidPatternIsSkipped() {
+    String patterns = "api/v[123,health,{unclosed";
+
+    assertThat(_filter.isExcluded("health", patterns)).isTrue();
+    assertThat(_filter.isExcluded("api/v1", patterns)).isFalse();
+  }
+
+  @Test
+  public void testInvalidPathHandling() {
+    String invalidPath = "path\0with\0nulls";
+    assertThat(_filter.isExcluded(invalidPath, "health")).isFalse();
+  }
+
+  @Test
+  public void testAllInvalidPatternsReturnFalse() {
+    String patterns = "[unclosed,{unclosed,\\invalid";
+
+    assertThat(_filter.isExcluded("anything", patterns)).isFalse();
+  }
+
+  // ===== Integration Test =====
+
+  @Test
+  public void testBasicIntegrationScenario() {
+    String patterns = "health,api/*,admin/**";
```

Comment:
nit: why is this marked as an integration test?

Reply:
updated.